captum/optim/_core/objectives.py

```diff
@@ -144,9 +143,9 @@ def optimize(
                 optimizer.step()
                 step += 1
         except (Exception, BaseException) as e:

```

Comment:
I don't think that we need to catch the exception now that `finally` is being used for `self.cleanup()`

Reply:
Oh yes that's true